Neos.ContentRepository.Export/src/Event/ValueObject/ExportedEvent.php

```diff
@@ -7,39 +7,39 @@
 final class ExportedEvent implements \JsonSerializable
 {
     /**
+     * The Neos.ContentRepository.Core's domain events require the payload to be a json array string
+     * {@see \Neos\ContentRepository\Core\EventStore\EventInterface}
+     * This exporter will enforce this as well (by using array as payload type).
+     *
      * @param array<mixed> $payload
      * @param array<mixed> $metadata
      */
     public function __construct(
         public readonly string $identifier,
         public readonly string $type,
-        public readonly array $payload, // TODO: string
+        public readonly array $payload,
         public readonly array $metadata,
-    ) {}
+    ) {
+    }
 
     public static function fromRawEvent(Event $event): self
     {
         return new self(
             $event->id->value,
             $event->type->value,
-            \json_decode($event->data->value, true),
+            \json_decode($event->data->value, true, 512, JSON_THROW_ON_ERROR),
             $event->metadata?->value ?? [],
         );
     }
 
     public static function fromJson(string $json): self
     {
         try {
-            ///** @var array{identifier: string, type: string, payload: array<mixed>, metadata: array<mixed>} $data */
-            /** @var array<mixed> $data */
+            /** @var array{identifier: string, type: string, payload: array<mixed>, metadata: array<mixed>} $data */
             $data = json_decode($json, true, 512, JSON_THROW_ON_ERROR);
         } catch (\JsonException $e) {
             throw new \InvalidArgumentException(sprintf('Failed to decode JSON "%s": %s', $json, $e->getMessage()), 1638432979, $e);
         }
-        assert(isset($data['identifier']) && is_string($data['identifier']));
-        assert(isset($data['type']) && is_string($data['type']));
-        assert(isset($data['payload']) && is_array($data['payload']));
-        assert(isset($data['metadata']) && is_array($data['metadata']));
```

Comment:
Why did you remove the assertions?
I'm fine with replacing `assert()` calls by actual assertions, but we should not just remove them IMO

Reply:
This should not happen as this is the assumed shape of the array.
And with the type annotations phpstan will even scream if we double check because we annotated that this is the shape.

Also in case there is a low level error a dedicated `TypeError` will be thrown which is much more explicit than this.

Reply:
yes, right, I overlooked that the types are assumed right below